pkg/kubelet/kubelet_node_status.go

```diff
@@ -418,9 +421,14 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 	if tryNumber == 0 {
 		util.FromApiserverCache(&opts)
 	}
-	node, err := kl.heartbeatClient.CoreV1().Nodes().Get(string(kl.nodeName), opts)
-	if err != nil {
-		return fmt.Errorf("error getting node %q: %v", kl.nodeName, err)
+
```

Comment:
please update the comments in line 413 to 419 for the new logic to avoid hitting the api server as possible and suffer a bit for cases where node was patched from other components under some situation.